### PR TITLE
fixed black screen bug when switch from background(Android)

### PR DESCRIPTION
**Pull request by 2439905184**
added isActive line

### CODE REVIEW

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 1]
```diff
@@ -127,6 +127,7 @@ public MainView(Context context) {
 		@Override
 		public void onSurfaceCreated(GL10 gl, EGLConfig config) {
 			// JNIコードで初期化処理を実行する
+			isActivie=true;
 			init();
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:131]: 1. isActivie --> isActive (spell)
2. The comment means "Do initialization in JNI code". Setting isActive is not in JNI code. This should be:
```
isActive = true;

// JNIコードで初期化処理を実行する
init();
```

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 2]
```diff
@@ -135,6 +136,7 @@ public void onSurfaceCreated(GL10 gl, EGLConfig config) {
 		 */
 		@Override
 		public void onSurfaceChanged(GL10 gl, int width, int height) {
+			isActivie=true;
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:139]: 1. isActivie --> isActive (spell)
2. Add spaces i.e. `isActive = true;`

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 3]
```diff
@@ -166,6 +168,11 @@ public void onDrawFrame(GL10 gl) {
 			if(isFinished)
 				return;
 
+			if(!isActivie)
+				return;
+
+
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:174]: 1. isActivie --> isActive (spell)
2. Remove empty lines

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 4]
```diff
@@ -223,6 +230,7 @@ else if(delta < -LINE_HEIGHT)
 	@Override
 	public void onPause() {
 		super.onPause();
+		isActivie=false;
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:233]: 1. isActivie --> isActive (spell)
2. Add spaces i.e. `isActive = false;`

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 5]
```diff
@@ -282,6 +290,9 @@ public void onResume() {
 	 */
 
 	/** 音声の再生を開始します。 */
+
+	private boolean isActivie=false;
+
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:295]: 1. isActivie --> isActive (spell)
2. Wrong position. The comment means "Start sound playback". This should be moved to:
```
    private boolean isFinished;

    /** Represents app is active or inactive. */
    private boolean isActive;
```
3. `=false;` is not needed. All boolean fields are set to false when instantiated.

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java [hunk 6]
```diff
@@ -242,7 +250,7 @@ public void onPause() {
 	@Override
 	public void onResume() {
 		super.onResume();
-
+		isActivie=true;
```
Comment (ktabata) [inline build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:253]: 1. isActivie --> isActive (spell)
2. Add spaces i.e. `isActive = false;`

**Review by ktabata — CHANGES_REQUESTED** [review]
Please check the review comments and change the code.

**Comment by ktabata** [comment]
On Android Emulator (Pixel 3a), I don't get black screen when I go to home screen and show Suika2 again.
Could you explain the steps to reproduce this bug?

**Comment by 2439905184** [comment]
> On Android Emulator (Pixel 3a), I don't get black screen when I go to home screen and show Suika2 again. Could you explain the steps to reproduce this bug?

on android true device .When I click mulit task button (system button ) and then click suika2 to return to run game again.Black screen will be showed. And there is sound still playing but not show image .

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java (hunk 1 above)
Comment (2439905184) [reply build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:131]: fixed

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java (hunk 2 above)
Comment (2439905184) [reply build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:139]: fixed

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java (hunk 3 above)
Comment (2439905184) [reply build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:174]: fixed

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java (hunk 4 above)
Comment (2439905184) [reply build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:233]: fixed

File: build/android/app/src/main/java/jp/luxion/suika/MainActivity.java (hunk 5 above)
Comment (2439905184) [reply build/android/app/src/main/java/jp/luxion/suika/MainActivity.java:295]: fixed

**Review by 2439905184 — COMMENTED** [review]
MainActivity.java is checked

**Review by 2439905184 — COMMENTED** [review]
[ fix: 修复退出游戏后再次进入黑屏的问题](https://github.com/2439905184/suika2/commit/5c0391085aecce12c54ef5ffd996ac29ed98beb4)
当按下返回键时询问是否确认退出游戏，避免误触。

**Comment by ktabata** [comment]
The problem was not in `MainActivity.java`, but `conf.c`.

Android reuses a process when a user reopens an app after close.
When a process is reused, global variables must be cleared to initial values.
In `conf.c`, I forgot to reset `loaded` flags on reuse.

Anyway, the change has already applied to the master branch.
I think we can close this PR.

**Comment by 2439905184** [comment]
ok just close it.

**Pull request closed without merging**